sw/device/lib/uart.c

```diff
@@ -42,5 +42,16 @@ void uart_send_uint(uint32_t n, int bits) {
 }
 
 int uart_rcv_char(char *c) {
-  return dif_uart_byte_receive_polled(&uart0, (uint8_t *)c);
+  int ret;
```

Comment:
You can early return, rather than having only a single `return` and using this variable.